### PR TITLE
Feature/im play

### PR DESCRIPTION
**Pull request by tonsV2**
This PR will make the pipeline deploy anything committed to master as an instance on IM.

The instance can be found here: https://play.im.prod.test.c.dhis2.org/dev

### CODE REVIEW

File: jenkinsfiles/dev [hunk 1]
```diff
@@ -182,6 +182,41 @@ pipeline {
...
+
+        stage('Update IM Play dev instance') {
+            if (env.GIT_BRANCH == 'master') {
+                steps {
+                    script {
+                        withCredentials([usernamePassword(credentialsId: 'e2e-im-user', passwordVariable: 'PASSWORD', usernameVariable: 'USER_EMAIL')]) {
+                            dir('im-db-manager') {
+                                sparseCheckout('https://github.com/dhis2-sre/im-database-manager', 'master', '/scripts')
+
+                                dir('/scripts') {
```
Comment (radnov) [inline jenkinsfiles/dev:194]: It should be just `scripts` here without the leading `/`.

File: jenkinsfiles/dev [hunk 2]
```diff
@@ -182,6 +182,41 @@ pipeline {
...
+
+                                    sh '[ -n "$DATABASE_ID" ]'
+                                    echo "DATABASE_ID is ${env.DATABASE_ID}"
+                                }
+                            }
+
+                            dir('im-manager') {
+                                sparseCheckout('https://github.com/dhis2-sre/im-manager', 'master', '/scripts')
+
+                                dir('im-manager/scripts') {
```
Comment (radnov) [inline jenkinsfiles/dev:209]: It should be just `scripts` here without the leading `im-manager/`, as we're already in the `im-manager` dir at that point.

File: jenkinsfiles/dev [hunk 3]
```diff
@@ -182,6 +182,41 @@ pipeline {
...
+                    script {
+                        withCredentials([usernamePassword(credentialsId: 'e2e-im-user', passwordVariable: 'PASSWORD', usernameVariable: 'USER_EMAIL')]) {
+                            dir('im-db-manager') {
+                                sparseCheckout('https://github.com/dhis2-sre/im-database-manager', 'master', '/scripts')
+
+                                dir('/scripts') {
+                                    env.DATABASE_ID = sh(
+                                            returnStdout: true,
+                                            // TODO: How do we avoid hard coding the name/version?
+                                            script: "./list.sh | jq -r '.[] | select(.Name == \"whoami\") | .Databases[] | select(.Name == \"Sierra Leone - 2.39.0.sql.gz\") | .ID'"
```
Comment (radnov) [inline jenkinsfiles/dev:198]: Name of the database should be something like “Sierra Leone - dev.sql.gz” (based on [the db manager seed script](https://github.com/dhis2-sre/im-database-manager/blob/master/scripts/seed.sh)); there’re no databases currently uploaded on prod, though.

File: jenkinsfiles/dev [hunk 4]
```diff
@@ -182,6 +182,41 @@ pipeline {
...
+                                    echo "DATABASE_ID is ${env.DATABASE_ID}"
+                                }
+                            }
+
+                            dir('im-manager') {
+                                sparseCheckout('https://github.com/dhis2-sre/im-manager', 'master', '/scripts')
+
+                                dir('im-manager/scripts') {
+                                    echo 'Creating DHIS2 instance ...'
+                                    sh "INSTANCE_HOST=https://api.im.prod.test.c.dhis2.org IMAGE_REPOSITORY=core-dev IMAGE_TAG=latest IMAGE_PULL_POLICY=Always ./deploy-dhis2.sh play dev"
```
Comment (radnov) [inline jenkinsfiles/dev:211]: The environment variables can be declared within an `environment` block, like [here for example](https://github.com/dhis2/dhis2-core/blob/master/jenkinsfiles/dev#L64-L69)

File: jenkinsfiles/dev [hunk 5]
```diff
@@ -182,6 +182,41 @@ pipeline {
                 }
             }
         }
+
+        stage('Update IM Play dev instance') {
+            if (env.GIT_BRANCH == 'master') {
+                steps {
+                    script {
+                        withCredentials([usernamePassword(credentialsId: 'e2e-im-user', passwordVariable: 'PASSWORD', usernameVariable: 'USER_EMAIL')]) {
```
Comment (radnov) [inline jenkinsfiles/dev:190]: Need to update that to `dhis2-im-bot` for the prod IM env.

**Review by radnov — CHANGES_REQUESTED** [review]

File: jenkinsfiles/dev (hunk 1 above)
Comment (tonsV2) [reply jenkinsfiles/dev:194]: How come it worked the other place? It's your code that I stole ;-)

File: jenkinsfiles/dev (hunk 2 above)
Comment (tonsV2) [reply jenkinsfiles/dev:209]: Same as above

File: jenkinsfiles/dev (hunk 3 above)
Comment (tonsV2) [reply jenkinsfiles/dev:198]: "Sierra Leone - 2.39.0.sql.gz" is the name of the database on the manager.

File: jenkinsfiles/dev (hunk 1 above)
Comment (radnov) [reply jenkinsfiles/dev]: It didn't :D You managed to copy it while it still was WIP and not tested

File: jenkinsfiles/dev [hunk 6]
```diff
@@ -182,6 +182,48 @@ pipeline {
...
+                HTTP = "http --check-status"
+                IM_HOST = "https://api.im.prod.test.c.dhis2.org"
+                INSTANCE_URL = "https://play.im.prod.test.c.dhis2.org/dev"
+                IMAGE_REPOSITORY = "core-dev"
+                IMAGE_TAG = "latest"
+                IMAGE_PULL_POLICY = "Always"
+            }
+            steps {
+                script {
+                    if (env.GIT_BRANCH == 'feature/im-play') {
```
Comment (radnov) [inline jenkinsfiles/dev:197]: Need to update this to `master`

**Review by radnov — CHANGES_REQUESTED** [review]
Looks good to me, just one tiny but important detail noted.

File: jenkinsfiles/dev (hunk 6 above)
Comment (tonsV2) [reply jenkinsfiles/dev:197]: Good point!

**Review by radnov — APPROVED** [review]

File: jenkinsfiles/dev [hunk 7]
```diff
@@ -182,6 +182,48 @@ pipeline {
...
+                                }
+                            }
+
+                            dir('im-manager') {
+                                sparseCheckout('https://github.com/dhis2-sre/im-manager', 'master', '/scripts')
+
+                                dir('scripts') {
+                                    echo 'Creating DHIS2 instance on IM...'
+                                    sh "(./findByName.sh play dev && ./restart.sh play dev) || ./deploy-dhis2.sh play dev"
+                                    waitFor.statusOk("${env.INSTANCE_URL}")
```
Comment (teleivo) [inline jenkinsfiles/dev:219]: How long will this wait? And I assume it will result in a failed step if it times out, right?

**Review by teleivo — COMMENTED** [review]
Nice 🤩 

How long does this stage take?

File: jenkinsfiles/dev (hunk 7 above)
Comment (tonsV2) [reply jenkinsfiles/dev]: It'll now timeout after 5 minutes

**Review by teleivo — APPROVED** [review]

**Pull request merged**